### DISCUSSION TITLE
STAC on Roland V-160HD

### DISCUSSION THREAD

**Discussion by feta00** [Xylopyrographer/STAC#55] [Ideas]
Hello, 
after trying to figure out why I keep getting a BPX (the big purple X as it is called on STAC documentation) I think I probably figured out why.

Smart tally on V-160HD works on a per INPUT base, not per CHANNEL.
When I open up a browser pointing the mixer IP address the web page asks me to choose an input to monitor (HDMI 1-8 or SDI 1-8), it doesn't allow me to monitor a channel. 
Selecting one of the inputs, let's say HDMI 1, the resulting web page is like this:
http://192.168.20.99/tally/hdmi_1

The same syntax is used for the other inputs, so for example for SDI 3, would be sdi_3.

I suspect that, since STAC was built upon V60-HD mixer, the smart tally on this unit works on a channel base (or a different input naming), so STAC keeps asking my V160 the status of a non-existing channel/input and then times-out. 
I can't check it on my own since we don't own one of these units, but I'd love to make STAC work on V-160HD and maybe could be useful for others.

Thanks for any feedback! 
And congratulations for the great work in creating STAC!

**Comment by Xylopyrographer** [comment]
@feta00 

I think it'd be awesome to get STAC working with the V-160HD! 

Would you be able to run a telnet session to pull some info?

In a terminal window could you do the following (assuming 192.168.20.99 is the IP address of your switcher and it's listening to http requests on port 80. Also hoping they haven't moved to `https://` 😬).

```
$ telnet 192.168.20.99 80

GET /index.html HTTP/1.1
```
Copy the entire session & paste it into a reply.

Hopefully something like this should come back (though the web page will be smushed into one very long line):

```
~ % telnet 192.168.20.99 80   <---- type this. may need to change '80' to the correct port.
Trying 192.168.20.99...
Connected to 192.168.20.99.
Escape character is '^]'.

GET /index.html HTTP/1.1      <---- type this. something like what follows should come back.

HTTP/1.1 200 OK
Connection: Keep-Alive
Content-Length: 1387
Content-Type: text/html; charset=ISO-8859-1
Server: Roland V-160HD

HTTP/1.1 200 OK
Connection: Keep-Alive
Content-Length: 1387
Content-Type: text/html; charset=ISO-8859-1
Server: Roland V-60HD

<!DOCTYPE html PUBLIC"-//W3C//DTD HTML 4.01//EN"><html lang="en">
<head>
<meta http-equiv="Content-Type"content="text/html; charset=iso-8859-1">
<script type="text/javascript">

var get_model=new XMLHttpRequest();
get_model.addEventListener('load',function(ev)
{
document.title='Roland '+get_model.response;
document.getElementById('model').textContent=get_model.response;
});
get_model.open('GET','/system/model');
get_model.send();

var get_npuver=new XMLHttpRequest();
get_npuver.addEventListener('load',function(ev){document.getElementById('npu_ver').textContent=get_npuver.response;
});
get_npuver.open('GET','/npu/version');
get_npuver.send();

var get_systemver=new XMLHttpRequest();
get_systemver.addEventListener('load',function(ev){document.getElementById('system_ver').textContent=get_systemver.response;
});
get_systemver.open('GET','/system/version');
get_systemver.send();

</script>
<style type="text/css">li{font-size:xx-large}
</style>
<title>Roland</title>
</head>
<body>
<ul>
<li>Model<ul>
<li id="model">
</li>
</ul>
</li>
<li>System Version<ul>
<li id="system_ver">
</li>
</ul>
</li>
<li>NPU Version<ul>
<li id="npu_ver">
</li>
</ul>
</li>
<li>Channel Status(Tally)<ul>
<li>
<a href="/tally/1">Ch.1</a>
</li>
<li>
<a href="/tally/2">Ch.2</a>
</li>
<li>
<a href="/tally/3">Ch.3</a>
</li>
<li>
<a href="/tally/4">Ch.4</a>
</li>
<li>
<a href="/tally/5">Ch.5</a>
</li>
<li>
<a href="/tally/6">Ch.6</a>
</li>
</ul>
</li>
</ul>
</body>
</html>
```

That should return the details needed to allow the STAC to query the right input.

Thanks.

**Reply by feta00** [reply]
Hi and thanks for the prompt reply. 
I will try asap and post the results.

**Comment by feta00** [comment]
hi, I wasn't able to run a telnet session for some reason I ignore, but I verified that http is being used and port 80 is open.
I'm attaching the html page source code, maybe is enough for you?
Let me know! Thanks.

[v-160.pdf](https://github.com/Xylopyrographer/STAC/files/14331131/v-160.pdf)

**Comment by Xylopyrographer** [comment]
@feta00

OK. Put together a test version for you in the [v160-dev](https://github.com/Xylopyrographer/STAC/tree/v160-dev) branch.

Check the `README.md` file for things you'll need to know first.

Pull down the code from there, give it a test & let me know what happens.

There is a `.bin` file [in there](https://github.com/Xylopyrographer/STAC/blob/v160-dev/bin/STAC_V160_dev03.bin) if you'll rather just use the OTA update feature of the STAC.

**Reply by feta00** [reply]
Hi, thanks for superfast feedback.
I gave it a try, installed STAC through ESP flasher using the bin file, all OK, could select one of the 16 channels but there we go again: BPX... So sad! Anything I can do to further investigate?

Thanks for your efforts.

**Comment by Xylopyrographer** [comment]
@feta00

Hi. I replied last week but seems that note disappeared. Apologies.

OK, we're very close. To get it working, need to capture a few transactions between the browser and the V-160HD after you select a channel to monitor. It appears Roland may have changed the tally status response message on that model.

Wonder if you could give a telnet season another go, as that way we can poke a message to the switch and check individual responses?

[Wireshark](https://www.wireshark.org) would also work, but that can be a bit of a bear to set up.

I'm reaching out to a few other sources to see if anyone else might be able to grab the transaction data as well.

**Reply by feta00** [reply]
Hi. I’ll sure try again and come back to you. 
Hope to have time today or tomorrow. 
Have a good day!

**Comment by feta00** [comment]
Hi, wireshark was a good advice, took a while to understand how it works but I finally managed to get some traffic out, see files attached.
10.254.52.200 is V-160HD IP address, 10.254.52.129 is my pc.
I'm attaching wireshark screenshot and two text files corresponding to traffic from switcher to pc and viceversa.


Hope this helps.

![Screenshot 2024-02-26 181415](https://github.com/Xylopyrographer/STAC/assets/107259076/2ba5f75d-65c8-4068-8903-fd076c274d66)
[Frame 63198 404 bytes on wire (3232.txt](https://github.com/Xylopyrographer/STAC/files/14407904/Frame.63198.404.bytes.on.wire.3232.txt)
[Frame 63191 60 bytes on wire (480 b.txt](https://github.com/Xylopyrographer/STAC/files/14407965/Frame.63191.60.bytes.on.wire.480.b.txt)

**Reply by Xylopyrographer** [reply]
This is great information. Glad you got WireShark up and running. 👍

Couple of questions:
1. Assume there was a browser running on the PC connected to the V-160HD?
2. Was the browser showing a state change (red, green, grey) as the switcher `HDMI 1` input was changed between `PGM`, `PVW` and `Unselected`?

It's interesting as the file "Frame.63191.60.bytes.on.wire.480.b.txt", the response from the V-160HD back to the PC, shows the "200" acknowledge code but it does not return the actual status information. I would expect to see one of `onair`, `selected` or `unselected`. This may be the "proprietary" part of the SmartTally protocol that Roland states. WireShark may be looking for fully formed HTTP transactions. (I think WireShark has a "raw capture" mode too.)

Would be interesting if you could run this again, using WireShark to monitor the communication between the v160-dev STAC and the V-160HD switch.

If that doesn't work, maybe you could try and get telnet working on the PC? It is not enabled by default on Windows. Quick article on how to enable it: [https://www.makeuseof.com/enable-telnet-windows/](https://www.makeuseof.com/enable-telnet-windows/)

We're almost there!

BTW, I've had the opportunity to travel a couple of times to Italy. Lovely people. Lovely country. Would love to return one day. Over here in Canada it's still winter. -17º C today, but forecast calls for it to be above 0 by the end of the week.

**Reply by feta00** [reply]
Hi, here the answers:
1. yes, there was chrome up and running connected to the switcher
2. the HDMI 1 channel was unselected at the time of the traffic analysis: I can give you the traffic onair and selected if it could be useful.

I will try tomorrow to monitor the communication between the stac and the switcher, I think Wireshark should work.
I don't know why telnet doesn't work, I already enabled it and tried both with cmd and putty: it always immediately closes the connection (or fails to open it)

BTW, never been to Canada unfortunately, glad you had a good time here! I'm in Trieste right know, it is worth a visit next time you come! And we could have a couple of beers too!

**Reply by feta00** [reply]
Hi, I tried again and you were right, the status information is present at the very end, see attached files!
I probably messed up something yesterday in copying data to the text files, sorry about that.

I also tried with the STAC connected, but it seems there is no communication at all between the STAC and the switcher, which seems quite weird, since STAC is connected to wifi and shows BPX.. what do you think? 
I keep trying..

[Frame 479 60 bytes on wire (480 bit.txt](https://github.com/Xylopyrographer/STAC/files/14418416/Frame.479.60.bytes.on.wire.480.bit.txt)
[Frame 1569 60 bytes on wire (480 bi.txt](https://github.com/Xylopyrographer/STAC/files/14418417/Frame.1569.60.bytes.on.wire.480.bi.txt)
[Frame 2654 60 bytes on wire (480 bi.txt](https://github.com/Xylopyrographer/STAC/files/14418418/Frame.2654.60.bytes.on.wire.480.bi.txt)

**Reply by Xylopyrographer** [reply]
All good.

At least the replies are somewhat consistent with the V-60HD.

The V-60HD will respond to "short" and "long" tally status requests. The STAC at the moment only uses the short form (expecting a short form response). You've managed to confirm that the V-160HD works on the long form for requests & replies. 👍 The BPX shown on the STAC is because it it expecting a short form response. (Bit more info in the _STAC Communications.md_ document.)

To test for short form requests and responses from the V-160HD, would need to get telnet working. Then you could poke short form requests to the switch and see what (if anything) comes back. The short form would be as described in the _STAC Communications.md_ except the request would be of the form `GET /tally/[channel]/status` where `channel` is `hdmi_n` or `sdi_n` with `n` between `1` and `8`.

I noticed this set of files has a different IP address for the switch. You have two V-160HD's? Probably worth checking, but assuming the STAC is configured to talk to the correct IP address?

**Reply by feta00** [reply]
no, we have one, but yesterday I just connected the switcher and pc to the lan in the venue I was working in.
Today I used the little AP I settled up to use with the STACs, that uses other IPs.. that's why they are different.

I wish I could get telnet working but no luck so far.
Could it be that the switcher does not support telnet? quite old protocol... maybe they cut it out.
I'll try again but I'm not very positive...

**Reply by Xylopyrographer** [reply]
Might be the switcher rejecting a telnet session. 🤔

As the STAC isn't sending stuff via a standard HTTP session (Roland's version) WireShark might nor pick it up using its HTTP protocol stack. I ca look in a bit to see if WS does indeed have a "raw" mode.

Appreciate the work you're doing.

**Reply by feta00** [reply]
Hi. Just found out on roland remote control manual that Telnet is enabled on V-160HD but uses port 8023, not 80. 
This might be the problem. 
I’ll try again tomorrow morning.

**Reply by Xylopyrographer** [reply]
Ah. Excellent 👍.

**Comment by feta00** [comment]
ok, so... one step forward.
Telnet connection ok, tried `GET /tally/[channel]/status` as you suggested (changing channel with one of the inputs), but I get `ERR0`, which appears to be a syntax error.

We did some research in [roland documentation](https://support.roland.com/hc/en-us/articles/4408252287515-V-160HD-Quickstart-Guide#h_01FQCTDRDV3ANQPP60H3XY684C) and made some testing using commands provided: the switcher replies correctly.

In particular this one example is interesting (you find it through the link above) and may be useful for our purposes:

> Example: Tally status of HDMI Input 3 on the crosspoint 
>
>If HDMI Input 3 is the PGM and/or PST selection, it will return a status value other than 0
>If HDMI Input 3 is not assigned to the crosspoint, the status value will always be 0
>Not the same as the tally light output, but may help control some third party tally systems
>
>RQH:0C0002,000001;
>Selects channel 3 on the PGM row of the Crosspoint
>
>RQH is the wrapper command
>0C 00 02 is the address of the HDMI Input 3 Tally Status SysEx command
>00 00 01 is the address size
>The control system receives the status as DTH:0C0002,0X 
>
>If HDMI Input 3 is selected as Nothing, X=0, PGM, X=1, PST, X=2, PGM & PST, X=3

We tried this RQH above using telnet and the switcher replies 0 if unselected, 1 if PGM etc.

All the codes for the inputs are in this [V-160HD_Control_eng04_W.pdf](https://github.com/Xylopyrographer/STAC/files/14433527/V-160HD_Control_eng04_W.pdf) at page 31. Other maybe useful info are at page 37.

Don't know if this could work in STACs or help us but I thought it could maybe be useful mentioning it.

**Reply by Xylopyrographer** [reply]
Hi. Interesting stuff for sure. Though I think that is all about controlling the V-160HD over a LAN connection. I'm guessing that port 8023 is for the remote control stuff, not the one used for the SmartTally feature. (Same port is used by the Roland RCS software.)

Just to confirm, trying to telnet into port 80 fails?

Did you try again to use Wireshark to monitor communications between the STAC and the V-160HD?

**Reply by feta00** [reply]
You’re probably right. 
Today I didn’t have enough time but I’ll try again tomorrow, both telnet and wireshark.

**Comment by feta00** [comment]
Hi, so I confirm that telnet to port 80 fails.

The only communication the STAC does is this ARP broadcast:
[ARP.txt](https://github.com/Xylopyrographer/STAC/files/14452026/ARP.txt)

Below you find the TCP and HTTP communications between the pc running smart tally web page and the switcher, the lines selected in the screenshot keeps repeating...

![Screenshot 2024-02-29 192528](https://github.com/Xylopyrographer/STAC/assets/107259076/65f7d1d4-e54a-480a-b232-dd5023d31ada)

 All selected lines are in this text file:
[ALL.txt](https://github.com/Xylopyrographer/STAC/files/14452048/ALL.txt)

I'm sorry I can't give you more infos but I noticed that the server inside the V-160HD is this so called [lwIP](https://savannah.nongnu.org/projects/lwip/) and I wonder if it is the same they put inside V-60HD... maybe not..

Thanks for your time, appreciated!

**Reply by Xylopyrographer** [reply]
This is actually quite interesting and useful. 👍

Can you confirm a couple things:
1. 192.168.2.99 is the IP address assigned to the STAC by the router?
1. The IP address of the switch is 192.168.2.101?
1. On the STAC, could you connect up a serial terminal to it, reset the STAC and send along a copy of the output? See as well the "Using screen for STAC Information.md" file.
1. Have you set the IP address of the V-160HD as "static" or "Manual" so it remains fixed?

The ARP thing is the STAC doing its thing to ID itself for mdns.

Let me noodle over things for a day or two & I'll get back to you.

**Reply by feta00** [reply]
No, the V-160HD IP is 192.168.20.99 and is fixed by a rule in router’s DHCP. 

192.168.20.101 is the pc where chrome browsing smart tally web page was up and running. 

The STAC IP is 192.168.20.100 and the only traffic it does is that ARP. 

I’ll try the serial terminal thing. 
By “reset” you mean “restart”?

**Reply by feta00** [reply]
hi, here the screen result... 
[Output di Terminale salvato.txt](https://github.com/Xylopyrographer/STAC/files/14457824/Output.di.Terminale.salvato.txt)

**Reply by Xylopyrographer** [reply]
OK. Thanks for the clarification and the screen dump.

Let me noodle on this for a day or two & I'll get back to you. 👍

**Reply by feta00** [reply]
Sure, no hurries, take your time..
Consider also I will be on vacation next week (March, 4th-12th) so I could read mails and stuff but won't be able to do any testing.

**Comment by feta00** [comment]
Hi, any news from your noodling? :-)
I'm back and ready to rock! 👍

**Reply by Xylopyrographer** [reply]
Hope the holiday was a good one.

I got wrapped up in a few things and didn't have quite the time I wanted to look at this but here is something I'd like to try:

Referring to [this article](https://osqa-ask.wireshark.org/questions/31856/how-to-capture-packets-between-2-ips/), could you use Wireshark to capture about 50 packets between a browser and the V-160HD. As the data flys back and forth pretty quick, probably need to do this three times with the channel being monitored in each of the program, preview and unselected states. Don't think it matters which channel is picked but it might be informative to do the group of three captures on the same channel and then do one on the other type. That is for example, a group of three captures for an SDI input and then a single capture on an HDMI input (or vice-versa). 

Could you then repeat this with another set of captures between the STAC V-160HD dev version and the V-160HD?

Wireshark can save the files in a "Wireshark compressed format". If you could then post all the files here, I should be able to open them up and poke about.

So that would be a total of eight captures in eight separate files.

Note that the short form of the the Roland Smart Tally protocol doesn't follow the standards for full http protocol which may be why the captures you did a couple weeks back looked incomplete as Wireshark tries to match by default the traffic flow based on the protocol.

Hope that was clear. Shoot me note with any questions. 👍

**Reply by feta00** [reply]
Hi, will try on Friday and give you a feedback. 
Thanks!

**Reply by feta00** [reply]
Hi, sorry if it took me so long.

So, find attached the scans between the V-160HD and the PC with the browser with smart tally running on it.
Tried to do the same between the switcher and the DEV version of the STAC but no packets at all are going on and forth between the two devices.

There is one more thing I think you should be aware of:
Roland recently released firmware version 3 of this unit, one thing has changed, they put in authentication when you try to reach the smart tally web page: username is "user" by deafult (and it probably can't be changed, and password is 0000 by deafult but you can change it if you like.

[scans.zip](https://github.com/Xylopyrographer/STAC/files/14864999/scans.zip)

**Reply by Xylopyrographer** [reply]
Thanks for the info. Good stuff in there. Need now some time to go through it.

Very odd that WS couldn't capture any data between the STAC and the V-160HD. I'll try and mimic that on our V-60 and let you know what I find.

I took a look at the Roland site and in their release notes for the [V-160HD System Program](https://proav.roland.com/global/support/by_product/v-160hd/updates_drivers/8534fcec-b377-418f-b103-41cd2d4f4d7e/) could not see any reference to Smart Tally. Would you have a reference to that? If they switched to https it'll take a bit to figure how to integrate that.

Should confirm: Are you running v2.x on your V-160HD? IF not, would need to capture the traffic from the start of when the web browser initiates the connection to the switch up to at least a couple of poll responses. (Maybe set the capture size to 100.)

**Reply by feta00** [reply]
Hi, I'm running v3.0.1 on my device, I updated the firmware since I really needed some of the new features, as you said there is no mention to smart tally in the release notes..
If you need another capture I could do that on Monday.

I believe anyway that they did not make any changes to smart tally and they did not switch to https (but I will check and come back to you); only thing they put up I the authentication I was mentioning..

**Reply by feta00** [reply]
Hi. I confirm they did not switch to HTTPS, still HTTP but with authentication. 
Let me know if you need another set of capture. Thanks

**Reply by Xylopyrographer** [reply]
Good news. Would you be able to do a WS capture from the start of the connection of a browser to the switch that includes the authentication request and to including a couple of tally status requests?

A screen shot of the Smart Tally browser authentication request would be handy too.

With v3.01 of the V-160HD software, how is the STAC responding?

FWIW, poking through the switcher docs, looks like the username and password are used only when you want to control the switcher via the LAN. Ref *V-160HD_Reference_eng06_W.pdf*, page 116.

**Reply by Xylopyrographer** [reply]
Took another look at the scan files you sent earlier. Think I figured out what is going on.

I'll try and get another version for you to test later this week.

No need to capture more info until after that (or not at all if it works!).

**Reply by feta00** [reply]
Great! Look forward to hear from you then!

**Comment by Xylopyrographer** [comment]
@feta00 

Good day. Release v160-dev_0.4 is available in the v160-dev branch. Binary is there as well. Load it up and give it a whirl after reading the notes in the README.md file, especially the requirement for the user and password in the LAN Control menu of the V-160HD. 

It would be useful to do a WS capture between a browser and the switch starting from when the browser first starts things by sending the IP address of the switch through to a couple of tally request transactions.

Looking at the Wireshark does, though it and the PC running it can be configured to monitor traffic not involving that PC, it is a rather arduous process and may not even be possible on a machine running Windows so we'll forego that.

A photo go the switch LAN Control menu would be useful as well.

Looking forward to what you find!

**Reply by feta00** [reply]
Hi. Thanks for your efforts, I will try as soon as possible, probably on Monday 15th, and come back to you.

**Reply by feta00** [reply]
Hi, so I gave it a go, here the results.

I still get a BPX on STAC, only difference with previous version is that it takes around 10 seconds to appear on the screen, while before it took maybe 1 second or 2. Maybe it's related to the login process...
I'm atttaching a short movie so you better understand.

I'm attaching another wireshark scan between the switcher and the browser, it begins with the login process and should include a couple of tally requests on HDMI1.
I tried a scan between the STAC and the switcher but again, no packets.

I'm also attaching a picture of the login on the browser and the LAN menu on the switcher.

Let me know if you need anything else.

[login_capture_hdmi1_pgm_pst_uns.zip](https://github.com/Xylopyrographer/STAC/files/14994367/login_capture_hdmi1_pgm_pst_uns.zip)

https://github.com/Xylopyrographer/STAC/assets/107259076/a75588ac-35eb-49a9-b9d2-a5b45a753db1

**Reply by Xylopyrographer** [reply]
Thanks for the great bunch of info and your tenacity in hanging in there while we figure this out.

So yes, the issue is getting past the authentication screen.

I'll work on another version that logs and dumps the traffic between the STAC and the V-160HD. Might take a bit to fit that in.

Interestingly, the WS capture didn't log the authentication transaction. That page would be of interest.

It looks like you're using Chrome? Could you get back to the V-160HD login screen and then in Chrome go to View--> Developer--> View Source and then paste the page here.

Also as a test, can the LAN user and password be disabled on the V-160HD? If so, could you retry the STAC after disabling that?

I'm very certain that this will work once we get the authentication sorted.

**Reply by feta00** [reply]
Hi, I tried to view the source of the login screen but chrome just opened another tab with the same webpage, no html code at all.
I tried with chrome developer tools, but all I get is a pretty empty html page with some css on it, that's it.
I'm attaching a small video describing this:

https://github.com/Xylopyrographer/STAC/assets/107259076/9b0629e9-01e4-4103-b333-052cfb0a4d17

What I noticed in the developer tools is this warning:
`One or more documents in this page is in Quirks Mode, which will render the affected document(s) with quirks incompatible with the current HTML and CSS specifications.
Quirks Mode exists mostly due to historical reasons. If this is not intentional, you can add or modify the DOCTYPE to be `<!DOCTYPE html>` to render the page in No Quirks Mode.`

I'm attaching the source code of the webpage that opens up after the login, but I doubt it would be useful.
And no, LAN user and password doesn't seem can be disabled. You can't even change the username, which has to be "user".

[view-source_192.168.20.99.zip](https://github.com/Xylopyrographer/STAC/files/15024687/view-source_192.168.20.99.zip)

**Comment by Xylopyrographer** [comment]
Quick update. Had other things that needed doing , back to looking at this again.

Should have another test version in a week or so.

**Reply by feta00** [reply]
yep, no problem! have a good day!

**Comment by Xylopyrographer** [comment]
@feta00 
Just posted dev-V0.5. A few things in the Release Notes to review, especially if you want t compile this version yourself.

There is a lot of debugging info sent the serial port with this one. Set the [polling interval to 1000 ms at first to see how it goes. 

In any case, please post the initial STAC serial port info dump and a few rounds of the polling cycle so I can see what's up.

**Reply by feta00** [reply]
Hi, I'm a bit in a hurry but just wanted to say that I just gave it a quick try and it works like a charm! 🥳

**Comment by Xylopyrographer** [comment]
Fantastic! 🍾 😄

When you do get a minute, the serial port data dumps would be useful.

I've uploaded now a binary version without the debugging code - it's in the /bin directory. Give that a try with the polling interval set to something around the default of 300 ms.

I'll let you run with that for a bit. There's is still some clean-up to do and then decide how to integrate this into the web setup pages and how to allow a user to pick the switch model when doing the configuration. Would appreciate any thoughts you might have, for example:

- allow min and max tally channel to be set by the user for the HDMI and SDI banks?
- allow a "user" name for the V-160HD network user name to be set?

Regarding the network password setting on the V-160HD: does it allow any character to be used in the password, or just numbers or ?? Does it have to be 4 characters or can it be from 0 to 4?

**Reply by feta00** [reply]
Hi, I will for sure post the serial port dumps as soon as I can.

Regarding the switcher selection, I'm not sure it would be better to have a single version of STAC with a switcher selection or two separate versions of STAC, one for each model.. I guess it depends on the amount of work required to integrate the V-160 version with the old V-60...

As for the tally channels, I think it would be better to have all the 16 channels always available to selection in the STACs, the selection is really easy and straightforward, so why make things more complicate than needed. Just my two cents of course...

As for the user, this can't be changed in the switcher, "user" is the only option available.
I'm quite sure the password has to be 4 numbers, and no other characters rather than numbers; I will check it anyway just to be sure.

**Comment by Xylopyrographer** [comment]
@feta00 : Just published a new release adding V-160HD support. Please give it a go and shoot back any feedback.
Major thanks for all the work supporting this release and of course for proposing the addition.

Closing with #56

**Reply by feta00** [reply]
Hi. Sorry, I’ve been very busy and could not find the time to test further as I wished. 
Will try asap but it could take me a couple of weeks. Sorry for that. 
And thank you for your work, patience and support!

**Reply by Xylopyrographer** [reply]
All good. Whenever you get around to it. 😄
